spec/index.bs

```diff
@@ -4095,6 +4118,123 @@ configuring each of the [=render stages=].
     the fragment shader entry point of the [=pipeline=] and its output colors.
     If it's `null`, the [[#no-color-output]] mode is enabled.
 
+### Primitive clipping ### {#primitive-clipping}
+
+Vertex shaders have to produce a built-in "position", which is a 4-component vector
+of floating-point values denoting the <dfn dfn>clip position</dfn> of a vertex.
+
+Primitives are clipped to the <dfn dfn>clip volume</dfn>, which, for any [=clip position=] |p|
+inside a primitive, is defined by the following set of restrictions:
+  - &minus;|p|.w &le; |p|.x &le; |p|.w
+  - &minus;|p|.w &le; |p|.y &le; |p|.w
+  - 0 &le; |p|.z &le; |p|.w (<dfn dfn>depth clipping</dfn>)
+
+If |descriptor|.{{GPURenderPipelineDescriptor/depthStencil}}.{{GPUDepthStencilState/clampDepth}} is `true`,
+the [=depth clipping=] restriction of the [=clip volume=] is not applied.
+
+A primitive passes through this stage unchanged if every one of its edges
+lie entirely inside the [=clip volume=].
+If the edges of a primitives intersect the boundary of the [=clip volume=],
```

Comment:
What if the edges of the primitive doesn't intersect the clip volume at all. Is it discarded?

Reply:
no, it's fine. It could still cover the whole screen without any particular edge intersecting the clip volume.